app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/fragments/DatasetListRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,91 @@
+package com.github.HumanLearning2021.HumanLearningApp.view.fragments
+
+import android.app.Activity
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.lifecycle.LifecycleCoroutineScope
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.recyclerview.widget.RecyclerView
+import com.github.HumanLearning2021.HumanLearningApp.R
+import com.github.HumanLearning2021.HumanLearningApp.model.DatabaseManagement
+import com.github.HumanLearning2021.HumanLearningApp.model.Dataset
+import com.github.HumanLearning2021.HumanLearningApp.model.DummyDatabaseManagement
+import kotlinx.coroutines.launch
+
+class DatasetListRecyclerViewAdapter(
+    private val hostActivity: Activity,
+    private val lifecycleScope: LifecycleCoroutineScope,
+    private val itemClickedCallback: (Dataset) -> Unit
+) : RecyclerView.Adapter<DatasetListRecyclerViewAdapter.ListItemViewHolder>() {
+
+    /**
+     * Defines the number of categories shown on one ListItemViewHolder
+     */
+    private val NB_REPRESENTATIVES_SHOWN = 3
+
+    private val dbMgt: DatabaseManagement = DummyDatabaseManagement.staticDummyDatabaseManagement
+
+    private val datasetList = MutableLiveData<List<Dataset>>()
+
+    init {
+        lifecycleScope.launch {
+            datasetList.value = dbMgt.getDatasets().toList()
+        }
+    }
+
+    /**
+     * Gets called when a new view holder is needed
+     * (for example when the user scrolls to reveal more items)
+     */
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListItemViewHolder {
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.fragment_dataset_list_elt, parent, false)
+        return ListItemViewHolder(view)
+    }
+
+    /**
+     * Gets called to allow the ViewHolder to be bound to its data, given its position
+     */
+    override fun onBindViewHolder(holder: ListItemViewHolder, position: Int) {
+        val ds = datasetList.value?.get(position)
+        holder.itemView.apply {
+            if (ds != null) {
+                findViewById<TextView>(R.id.DatasetList_elt_text).text = ds.name
+
+                setRepresentativePictures(this, ds)
+            }
+        }
+        // so that the holder can call the callback with the correct dataset
+        holder.bindDataset(ds!!) // should be safe because position has to be  in {0..size-1}
```

Comment:
comment is a bit confusing given that the previous bit of code was null-checked. If we assume that datasetList.value isn't null you might as well put a double bang at line 54. Although with that said it still looks like a race condition with line 36... Maybe you meant to alias `mutableDatasetList` from the enclosing fragment?

Reply:
Indeed there are some problems in this part. Will look into it

Reply:
I think I found a good solution, will push soon